substratevm/DebugInfo.md

```diff
@@ -76,12 +80,83 @@ created during population of the cache.
 Note that in all the examples above the `DebugInfoSourceSearchPath`
 options are actually redundant. In the first case, the classpath
 entries for _apps/hello/classes_ and _apps/greeter/classes_ will be used
-to derive the default search roots `apps/hello/src` and
+to derive the default search roots _apps/hello/src_ and
 _apps/greeter/src_. In the second case, the classpath entries for
 _apps/target/hello.jar_ and _apps/target/greeter.jar_ will be used to
 derive the default search roots _apps/target/hello-sources.jar_ and
 _apps/target/greeter-sources.jar_.
 
+## Debugging with Isolates
+
+Note that it is currently recommended to disable use of Isolates by
+passing flag `-H:-UseIsolates` on the command line when debug info
+generation is enabled. Enabling of Isolates affects the way that oops
+(object references) are encoded. In turn that means the debug info
+generator has to provide gdb with information about how to translate
+an encoded oop to the address in memory where the object data is
+stored. This sometimes requires care when asking gdb to process
+encoded oops vs decoded raw addresses.
+
+When isolates are disabled oops are essentially raw addresses pointing
+directly at the object contents. This is the same whether the oop is
+stored in a static/instance field or has been loaded into a register.
+
+When an oop is stored in a static or instance field gdb knows the type
+of the value stored in the field and knows how to dereference it to
+locate the object contents. For example, assume we have a `Units` type
+that details the scale used for a blueprint drawing and that the
+`Units` instance has a `String` field called `print_name`. Assume also
+we have a static field `DEFAULT_UNIT` that holds the standard `Units`
+instance. The following command will print the name for the default
+units.
+
+```
+(gdb) print *com.acme.Blueprint::DEFAULT_UNIT->print_name
```

Comment:
Static field access like this one should be tested in `testhello.py`

Reply:
Yes, good point. I'll add a check to the test.